rollup/internal/controller/relayer/l1_relayer.go

```diff
@@ -150,6 +154,15 @@ func (r *Layer1Relayer) ProcessGasPriceOracle() {
 		}
 
 		if r.shouldUpdateGasOracle(baseFee, blobBaseFee, isCurie) {
+			// if we are not committing batches due to high fees then we shouldn't update fees to prevent users from paying high l1_data_fee
+			// also, set fees to some default value, because we have already updated fees to some high values, probably
+			if r.commitBatchReachTimeout() {
```

Comment:
It's nice to have a unit test covering this path (update gas oracle price with the default `basefee` and `blobbasefee`).

Reply:
How about add `failure_type` to `PendingTransaction`? when the sender fails, record the failure type. 

So, `commitBatchReachTimeout` reached, and also check the latest `PendingTransaction` failure type, if the failure type is `HighFee`, then reset the `baseFee, blobBaseFee`

Reply:
I added a unit test
I am not sure if it is possible to determine that transaction failed due to high fees. If I understand correctly:
if tx is not confirmed during some time it's fee just increased and tx resubmitted.